doc/man3/SSL_key_update.pod

```diff
@@ -0,0 +1,119 @@
+=pod
+
+=head1 NAME
+
+SSL_key_update,
+SSL_get_key_update_type,
+SSL_renegotiate,
+SSL_renegotiate_abbreviated,
+SSL_renegotiate_pending
+- initiate and obtain information about updating connection keys
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ /* TLSv1.3 KeyUpdate message types */
+ typedef enum {
+     /* -1 used so that this is an invalid value for the on-the-wire protocol */
+     SSL_KEY_UPDATE_NONE = -1,
+     /* Values as defined for the on-the-wire protocol */
+     SSL_KEY_UPDATE_NOT_REQUESTED = 0,
+     SSL_KEY_UPDATE_REQUESTED = 1
+ } SSL_KEY_UPDATE;
+
+ int SSL_key_update(SSL *s, SSL_KEY_UPDATE updatetype);
+ SSL_KEY_UPDATE SSL_get_key_update_type(SSL *s);
+
+ int SSL_renegotiate(SSL *s);
+ int SSL_renegotiate_abbreviated(SSL *s);
+ int SSL_renegotiate_pending(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_key_update() schedules an update of the keys for the current TLS connection.
+If the B<updatetype> parameter is set to B<SSL_KEY_UPDATE_NOT_REQUESTED> then
```

Comment:
I guess I should make these "#define" values now, given that we're not using an enum anymore. The UPDATE_REQUESTED and UPDATE_NOT_REQUESTED names are what they are called in the RFC so I thought we should maintain consistency.